### DISCUSSION TITLE
Slim crop uploader - saving output file as IM resized image and thumb

### DISCUSSION THREAD

**Discussion by darren0457** [ImageMagick/ImageMagick#2059] [Help]
Hi 

How do I go about having an Imagemagick command resize an image before its saved?

Im looking to save a 400px & 100px thumb to additional folders with an IM command so something along the lines of:

exec ("convert $imgage -resize 400x400  ??

***************************************************************
This is sync.php file from slim uploader
***************************************************************
require_once('slim.php');
// Get posted data
$images = Slim::getImages();
if ($images == false) {
    // inject your own auto crop or fallback script here
    echo '<p>Slim was not used to upload these images.</p>';
}
else {
// Could be multiple slim croppers
    foreach ($images as $image) {

        $files = array();
        // save output data if set
        if (isset($image['output']['data'])) {

            // Save the file
            $name = $image['output']['name'];

            // We'll use the output crop data
            $data = $image['output']['data'];	
										
            // Full size image for printing - 4000px - ok works no issues
            $file = Slim::saveFile($data, $name, '../myfolderA/'); 

             // need IM to resize to 400px & save folderB
             $output = Slim::saveFile($data, $name, '/folderB/'); 

              // need thumb at 100px 
             $new = Slim::saveFile($data, $name, '/folderC/'); 

            array_push( $files, $output);
}
$filenames = join(', ', array_map(function($file){ return ellipsis($file['name'], 100); }, $files));
        $images = array_map(function($file) { return '<img src="' . $file['path'] . '" alt=""/>'; }, $files);
    }
}
function ellipsis($str, $len = 50) {
    return strlen($str) > $len ? substr($str, 0, $len) . '...' : $str;
}

***************************************************************

Any help greatly appreciated

**Comment by snibgo** [comment]
I don't know what "$imgage" is. But an IM command to read a file, resize the image, and overwrite the file would be:

`magick infile -resie 400x400 infile`

If you use IM v6, use "convert" instead of "magick".
